### PR TITLE
added tensor device conversion for solarize params

### PR DESCRIPTION
**Pull request by mshalvagal**
### Description
Resolves issue #622 
([Bug] Solarize transform breaks with cuda input tensors)

Added statements to force solarize params to same device and type as input tensor.

### Status
**Ready**

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/enhance/adjust.py [hunk 1]
```diff
@@ -297,6 +298,8 @@ def solarize(input: torch.Tensor, thresholds: Union[float, torch.Tensor] = 0.5,
             # TODO: I am not happy about this line, but no easy to do batch-wise operation
             additions = torch.stack([x.expand(*input.shape[1:]) for x in additions])
 
+        additions = additions.to(input.device).to(input.dtype)
+
```
Comment (shijianjian) [inline kornia/enhance/adjust.py:302]:
```suggestion
            additions = additions.to(input.device).to(input.dtype)
            additions = torch.stack([x.expand(*input.shape[1:]) for x in additions])
```
Same above.

No additional indent here or it might throw an error if ```additions``` is a float number.

File: kornia/enhance/adjust.py [hunk 2]
```diff
@@ -250,6 +250,7 @@ def _solarize(input: torch.Tensor, thresholds: Union[float, torch.Tensor] = 0.5)
             f"threshholds must be a 1-d vector of shape ({input.size(0)},). Got {thresholds}"
         # TODO: I am not happy about this line, but no easy to do batch-wise operation
         thresholds = torch.stack([x.expand(*input.shape[1:]) for x in thresholds])
+        thresholds = thresholds.to(input.device).to(input.dtype)
```
Comment (shijianjian) [inline kornia/enhance/adjust.py:253]:
```suggestion
        thresholds = thresholds.to(input.device).to(input.dtype)
        thresholds = torch.stack([x.expand(*input.shape[1:]) for x in thresholds])
```
It is better to make the calculation happens inside your pointed device as well.

Comment (edgarriba) [reply kornia/enhance/adjust.py:253]: Also `thresholds = thresholds.to(input)` will work

File: kornia/enhance/adjust.py (hunk 1 above)
Comment (mshalvagal) [reply kornia/enhance/adjust.py:302]: The problem is `additions` is cast to a tensor even in case it's a float, which is how I ran into this issue in the first place. So, either we remove that line (link below) or we can always treat it as a tensor.

[https://github.com/kornia/kornia/blob/ec0cf87659e4c8f168ddacbabf714a1d555b29e9/kornia/enhance/adjust.py#L281-L282](url)
